substrate/frame/salary/src/lib.rs

```diff
@@ -382,6 +391,63 @@ pub mod pallet {
 		}
 	}
 
+	#[pallet::tasks_experimental]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		#[pallet::task_list({
+			let now = frame_system::Pallet::<T>::block_number();
+			let status = Status::<T, I>::get().expect("Task require active status");
+			let cycle_period = Pallet::<T, I>::cycle_period();
+			let result = if now >= status.cycle_start + cycle_period {
+				vec![()] // Success one task available: `()`.
+			} else {
+				vec![] // Failure no task available.
+			};
+			result.into_iter()
```

Comment:
we shouldn't do `expect` here.
```suggestion
			if let Some(status) = Status::<T, I>::get() {
				let now = frame_system::Pallet::<T>::block_number();
				let cycle_period = Pallet::<T, I>::cycle_period();
				if now >= status.cycle_start + cycle_period {
					vec![()] // Success one task available: `()`.
				} else {
					vec![] // Failure no task available.
				}
				result.into_iter()
			} else {
				vec![].into_iter() // No task available.
			}
```

I would prefer to use some `let Some(status) = ... else { return ... }` but this macro doesn't encapsulate the expression into a function :-/

Reply:
A way to add some context between the failing outputs would be nice, not sure how that would be useful to the node operators?

Reply:
I think it is fine, if the status is none, it means it hasn't started I think. So there is no bump to do.